### PR TITLE
Change setter & getter lattice interface

### PR DESCRIPTION
**Pull request by murilobalves**

### CODE REVIEW

File: pyaccel/lattice.py [hunk 1]
```diff
@@ -559,15 +541,14 @@ def set_error_rotation_roll(lattice, indices, values):
...
-        cos, sin = _math.cos(val), _math.sin(val)
-        rot = _numpy.diag([cos, cos, cos, cos, 1.0, 1.0])
-        rot[0, 2], rot[1, 3], rot[2, 0], rot[3, 1] = sin, sin, -sin, -sin
-
-        for idx in segs:
+    for segs, vals in zip(indices, values):
+        for idx, val in zip(segs, vals):
+            cos, sin = _math.cos(val), _math.sin(val)
+            rot = _numpy.diag([cos, cos, cos, cos, 1.0, 1.0])
+            rot[0, 2], rot[1, 3], rot[2, 0], rot[3, 1] = sin, sin, -sin, -sin
```
Comment (fernandohds564) [inline pyaccel/lattice.py]: Bug here. There's no double loop in the original method!

File: pyaccel/lattice.py [hunk 2]
```diff
@@ -601,15 +582,15 @@ def add_error_rotation_roll(lattice, indices, values):
...
 
     # loops over elements and sets its R1 and R2 fields
-    for segs, val in zip(indices, values):
-        cos, sin = _math.cos(val), _math.sin(val)
-        rot = _numpy.diag([cos, cos, cos, cos, 1.0, 1.0])
-        rot[0, 2], rot[1, 3], rot[2, 0], rot[3, 1] = sin, sin, -sin, -sin
+    for segs, vals in zip(indices, values):
 
-        for idx in segs:
+        for idx, val in zip(segs, vals):
```
Comment (fernandohds564) [inline pyaccel/lattice.py]: Same as before.

File: pyaccel/lattice.py [hunk 3]
```diff
@@ -528,17 +512,15 @@ def get_error_rotation_roll(lattice, indices):
...
     # loops over elements and gets error from R_IN
     values = []
     for segs in indices:
         for idx in segs:
             angle = _math.asin(lattice[idx].r_in[0, 2])
             values.append(angle)
-    if len(values) == 1:
-        return values[0]
-    return values
+    return _process_output(values, isflat)
```
Comment (fernandohds564) [inline pyaccel/lattice.py]: There's a bug here. Remember that values must have the same structure as indices. It must be a list of lists.

File: pyaccel/lattice.py [hunk 4]
```diff
@@ -675,26 +654,27 @@ def set_error_rotation_pitch(lattice, indices, values):
...
-
-        # Apply the errors only to the entrance of the first and exit of the
-        # last segment:
-        lattice[segs[0]].t_in[2] = -(L/2)*angy + misy
-        lattice[segs[-1]].t_out[2] = -(L/2)*angy - misy
-        lattice[segs[0]].t_in[3] = angy
-        lattice[segs[-1]].t_out[3] = -angy
-        lattice[segs[-1]].t_out[5] = path
+    for segs, angs in zip(indices, values):
+        for ang in angs:  # Doubt!!!
```
Comment (fernandohds564) [inline pyaccel/lattice.py]: We talk about this method on Tuesday.

File: pyaccel/lattice.py (hunk 1 above)
Comment (murilobalves) [reply pyaccel/lattice.py]: corrected

File: pyaccel/lattice.py (hunk 2 above)
Comment (murilobalves) [reply pyaccel/lattice.py]: corrected

File: pyaccel/lattice.py (hunk 3 above)
Comment (murilobalves) [reply pyaccel/lattice.py]: fixed

File: pyaccel/lattice.py (hunk 4 above)
Comment (murilobalves) [reply pyaccel/lattice.py]: done

**Pull request merged**